src/viam/sdk/rpc/dial.cpp

```diff
@@ -35,10 +35,16 @@ void ViamChannel::close() {
         return;
     }
     closed_ = true;
+    // CR erodkin: every now and again this call causes a failure. From adding log statements
+    // in rust-utils, I have confirmed that the failure happens within the rust code.
```

Comment:
Flagging this, for some reason `free_string` very occasionally fails in the rust call, specifically [here](https://github.com/viamrobotics/rust-utils/blob/6a7f896403d99d2f26e6464a3d86fa42eeb4615e/src/ffi/dial_ffi.rs#L285). Because the rust code is so minimal I haven't tried much on that side beyond explicitly dropping the `CString`, which didn't seem to make a difference.